invoices/invoices.go

```diff
@@ -471,6 +471,24 @@ func (i *Invoice) HTLCSetCompliment(setID *[32]byte,
 	return htlcSet
 }
 
+// IsKeysend returns true if the invoice is a Keysend invoice.
+func (i *Invoice) IsKeysend() bool {
```

Comment:
Yeah this is how it's detected elsewhere: https://github.com/lightningnetwork/lnd/blob/c38be044d362f3cb3d9ca602de15115f70c5bad4/invoices/invoice_expiry_watcher.go#L239

Optionally, we can also check the custom records for the keysend type.